### DISCUSSION TITLE
BUG!!!The webui controlnet plugin's masked upload feature is not working at all, or rather it's working in a strange way.

### DISCUSSION THREAD

**Discussion by qingweng123** [Mikubill/sd-webui-controlnet#2915] [Q&A]
<img width="1022" alt="屏幕截图 2024-05-20 162452" src="https://github.com/Mikubill/sd-webui-controlnet/assets/170290821/37a72d13-a861-4917-ac13-35e6dd614f10">
<img width="607" alt="屏幕截图 2024-05-20 162533" src="https://github.com/Mikubill/sd-webui-controlnet/assets/170290821/ec249c1d-78de-41f9-b00d-34d4dd0e94ff">
First of all, the text of the raw image controlnet inpaint (local repaint) no matter how you upload the black and white mask, it does not work, that is, the black area does not block the effect of inpaint, the white area does not work the effect of inpaint, and even in the generation of the result is not a black and white mask, either black and white mask to play the shape of the control and so on.
    <img width="1025" alt="屏幕截图 2024-05-20 162950" src="https://github.com/Mikubill/sd-webui-controlnet/assets/170290821/bc4eb819-62d6-4ca0-9cbf-51d2554d43bd">
I see that the upload mask function in
 controlnet developed by the big guy is for inpaint and ip adapter, currently my controlnet plugin version is the latest, but the upload mask function doesn't work for inpaint and ip adpater, instead, the canvas graffiti mask in inpaint in controlnet is useful for inpaint works
<img width="1004" alt="屏幕截图 2024-05-20 163316" src="https://github.com/Mikubill/sd-webui-controlnet/assets/170290821/31e13d51-bf1e-4017-81b9-eccd01002545"> 
The forge webui has no problem uploading masks, but the normal webui doesn't work at all!

**Comment by mmguero** [comment]
Are you thinking to submit an issue or PR for this? I can submit an issue and reference this and my other comment if you're not planning on doing so.

**Comment by huchenlei** [comment]
See pinned issue: https://github.com/Mikubill/sd-webui-controlnet/issues/2829

**Reply by mmguero** [reply]
Not sure I understand, @huchenlei. Are you saying #2829 addresses the [issue I have](https://github.com/AUTOMATIC1111/stable-diffusion-webui/discussions/15699) and that @c0ffincolors has mentioned above? I see that the issue mentions "Replace mask upload with effective region mask," but was that intended to break/remove the old functionality of being able to inpaint just a specific area with controlnet?

For what it's worth I patched my controlnet.py locally with the change @c0ffincolors mentioned [above](https://github.com/Mikubill/sd-webui-controlnet/discussions/2915#discussioncomment-9733172) and that did fix the issue for me, or at least made effective region mask work how I expected it to with controlnet inpaint (e.g., only changing the white areas of the mask and leaving the black areas alone).
